File: project 3/part II _ C++/task2.cpp
#include <iostream>
using namespace std;

bool binarySearch(int list[], int value, int min, int max ){
    // print the list inputs
    cout << "value: "<<value<<" min: "<<min<<"  max: "<<max<< endl;
    
    
    // first check condition that value is not in list
    if(min>max){
        cout << "Your value  "<<value<<"  is not in this list sorry "<< endl;
        return false;
    }
    else{
        int center=(int)((max+min)/2);
        if(list[center]<value){
            // search the right side
            binarySearch(list, value, center, max); // recursion
        }
        else if(list[center]>value){
            // search left side
            binarySearch(list, value, min, center); // recursion
        }
        else{
            cout << "We found your value : "<<value<< endl;
            return true;
        }
    }
}

int main(){
   cout << "Hello Mkhanyisi lets search for your value!!!" << endl; 
   
   int n[ 100 ]; //  an array of 100 integers
 
   // initialize elements of array n to 0          
   for ( int i = 0; i < 100; i++ ) {
      n[ i ] = i ;
   }
   
   binarySearch(n, 66,0,99);
   
   return 0;
}

File: project 3/Part II _ Ruby/task2.rb
#  Ruby demonstration Mkhanyisi 
puts "Binary search ";


$g_variable = 333 # declaring a global variable

def binarySearch(list ,value,min ,max )
    @value=value
    @min=min
    @max=max
    puts "searching for #@value in the list of numbers from #@min -  #@max"
    if(min>max) # terminating condition, variable not there
        puts ">value variable #@value not in list";
        return -1
    else
        @center=((max+min)/2).to_i
        
        if(list[@center]<value) # right side
            return  binarySearch(list ,value,@center+1 ,max )
        elsif(list[@center]>value) # left side
            return binarySearch(list ,value,min ,@center+1 )
        else # object found 
            puts ">found your value #@value"
            return 1
        end
    end
end


$digits = Array(0..67)

$output=binarySearch($digits ,22,0 ,67 )


puts "done #$output"



